x/evm/keeper/vote_handler.go

```diff
@@ -0,0 +1,162 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/axelarnetwork/utils/slices"
+
+	"github.com/axelarnetwork/axelar-core/x/evm/types"
+	nexus "github.com/axelarnetwork/axelar-core/x/nexus/exported"
+	vote "github.com/axelarnetwork/axelar-core/x/vote/exported"
+)
+
+// NewVoteHandler returns the handler for processing vote delivered by the vote module
+func NewVoteHandler(cdc codec.Codec, keeper types.BaseKeeper, nexus types.Nexus) vote.VoteHandler {
+	return func(ctx sdk.Context, result *vote.Vote) error {
+		events, err := types.UnpackEvents(cdc, result.Results)
+		if err != nil {
+			return err
+		}
+
+		if len(events) == 0 {
+			return nil
+		}
+
+		chainName := events[0].Chain
+		if slices.Any(events, func(event types.Event) bool { return event.Chain != chainName }) {
+			return fmt.Errorf("events are not from the same source chain")
+		}
+
+		chain, ok := nexus.GetChain(ctx, chainName)
+		if !ok {
+			return fmt.Errorf("%s is not a registered chain", chainName)
+		}
+
+		var errors []error
+		for _, event := range events {
+			var err error
+
+			// validate event
+			err = event.ValidateBasic()
+			if err != nil {
+				errors = append(errors, fmt.Errorf("event %d: %s", event.Index, err.Error()))
+				continue
+			}
+
+			// check if event confirmed before
+			eventID := event.GetID()
+			if _, ok := keeper.ForChain(chainName).GetEvent(ctx, eventID); ok {
+				errors = append(errors, fmt.Errorf("event %s is already confirmed", eventID))
+				continue
+			}
+
+			switch event.GetEvent().(type) {
```

Comment:
```suggestion
			switch event:= event.GetEvent().(type) {
```
then you don't have to cast it inside the handle funcs